src/workers/newView.ts

```diff
@@ -24,12 +24,78 @@ const addView = async (con: DataSource, entity: View): Promise<boolean> => {
   return false;
 };
 
+const DEFAULT_TIMEZONE = 'UTC'; // in case user doesn't have a timezone set
+const INC_STREAK_QUERY = `
+WITH u AS (
+  SELECT "id",
+      /* Increment the current streak if
+        a) lastViewAt is NULL - this is a new user, we should start a new streak
+        b) the currentStreak is 0 - we should start a new streak
+        c) we didn't do it today already
+      */
+      ("lastViewAt" is NULL OR
+       "currentStreak" = 0 OR
+       DATE("lastViewAt" AT TIME ZONE COALESCE("timezone", $3)) <> DATE($2 AT TIME ZONE COALESCE("timezone", $3))
+      ) AS shouldIncrementStreak,
+
+      -- Increment maxStreak if it's the same as currentStreak
+      "currentStreak" >= "maxStreak" AS shouldUpdateMaxStreak
+  FROM public.user AS users
+  INNER JOIN user_streak ON user_streak."userId" = users.id
+  WHERE users.id = $1
+)
+UPDATE user_streak AS us SET
+  "lastViewAt" = $2,
+  "updatedAt" = now(),
+  "currentStreak" = CASE WHEN shouldIncrementStreak THEN us."currentStreak" + 1 ELSE us."currentStreak" END,
+  "totalStreak" = CASE WHEN shouldIncrementStreak THEN us."totalStreak" + 1 ELSE us."totalStreak" END,
+  "maxStreak" = CASE WHEN shouldIncrementStreak AND shouldUpdateMaxStreak THEN us."maxStreak" + 1 ELSE us."maxStreak" END
```

Comment:
Just thinking this will be more consistent:
```suggestion
  "maxStreak" = GREATEST("maxStreak", us."currentStreak" + 1)
```

Reply:
Changed. We cannot use GREATEST like you suggest unfortunately, because we only want to increment maxStreak if we are doing the currentStreak increment as well.

Reply:
You're right! We can add `CASE WHEN shouldIncrementStreak` in this case. My biggest concern is to have drifts. This is why I prefer to have the logic sticked to the query and reduce the assumptions here. Does that make sense?